provider/equinix/environ.go

```diff
@@ -0,0 +1,665 @@
+			Public:        false,
+			AddressFamily: 4,
+			CIDR:          31,
+		})
+	}
+	if !requestedPublicAddr {
+		// Allocate a public address from the default address pool.
+		device.IPAddresses = append(device.IPAddresses, packngo.IPAddressCreateRequest{
+			Public:        true,
+			AddressFamily: 4,
+			CIDR:          31,
+		})
+	}
+
+	d, _, err := e.equnixClient.Devices.Create(device)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	d, err = waitDeviceActive(e.equnixClient, d.ID)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	inst := &equnixDevice{e, d}
+	amd64 := arch.AMD64
+	mem, err := strconv.ParseUint(d.Plan.Specs.Memory.Total[:len(d.Plan.Specs.Memory.Total)-2], 10, 64)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	var cpus uint64 = 1
+	if inst.Plan != nil && inst.Plan.Specs != nil && len(inst.Plan.Specs.Cpus) > 0 {
+		cpus = uint64(inst.Plan.Specs.Cpus[0].Count)
+	}
+
+	return &environs.StartInstanceResult{
+		Instance: inst,
+		Hardware: &instance.HardwareCharacteristics{
+			Arch: &amd64,
+			Mem:  &mem,
+			// RootDisk: &instanceSpec.InstanceType.RootDisk,
+			CpuCores: &cpus,
+		},
+	}, nil
+}
+
+func (e *environ) getSubnetsToZoneMap(ctx context.ProviderCallContext, args environs.StartInstanceParams) ([]string, error) {
+	var subnetIDs []string
+	for _, subnetList := range args.SubnetsToZones {
+		for subnetID := range subnetList {
+			packetSubnetID := strings.TrimPrefix(subnetID.String(), "subnet-")
+			subnetIDs = append(subnetIDs, packetSubnetID)
+		}
+	}
+
+	return subnetIDs, nil
+}
+
+// supportedInstanceTypes returns the instance types supported by Equnix Metal.
+func (e *environ) supportedInstanceTypes() ([]instances.InstanceType, error) {
+	opt := &packngo.ListOptions{
+		Includes: []string{"available_in"},
+	}
+	plans, _, err := e.equnixClient.Plans.List(opt)
+	if err != nil {
+		return nil, errors.Annotate(err, "retrieving supported instance types")
+	}
+
+	var instTypes []instances.InstanceType
+nextPlan:
+	for _, plan := range plans {
+		if !validPlan(plan, e.cloud.Region) {
+			logger.Infof("Plan %s not valid in facility %s", plan.Name, e.cloud.Region)
+			continue
+		}
+
+		var instArch string
+		switch {
+		case strings.HasSuffix(plan.Name, ".x86"):
+			instArch = arch.AMD64
+		case strings.HasSuffix(plan.Name, ".arm"):
+			instArch = arch.ARM64
+		default:
+			continue nextPlan
+
+		}
+
+		mem, err := parseMemValue(plan.Specs.Memory.Total)
+		if err != nil {
+			continue
+		}
+
+		instTypes = append(instTypes,
+			instances.InstanceType{
+				Id:       plan.ID,
+				Name:     plan.Name,
+				CpuCores: uint64(plan.Specs.Cpus[0].Count),
+				Mem:      mem,
+				Arches:   []string{instArch},
+				// Scale per hour costs so they can be represented as an integer for sorting purposes.
+				Cost: uint64(plan.Pricing.Hour * 1000.0),
+				// TODO: returned by packet's API but not exposed by the packngo client
+				// Deprecated: plan.Legacy,
```

Comment:
@displague Any progress on adding this particular field to the client?

Reply:
Legacy is exposed - https://github.com/packethost/packngo/blob/master/plans.go#L25 in packngo 0.13.0